src/sentry/dynamic_sampling/prioritise_transactions.py

```diff
@@ -32,10 +32,49 @@
 QUERY_TIME_INTERVAL = timedelta(hours=1)
 
 
+class ProjectIdentity(TypedDict, total=True):
+    """
+    Project identity, used to match projects and also to
+    order them
+    """
+
+    project_id: int
+    org_id: int
+
+
 class ProjectTransactions(TypedDict, total=True):
+    """
+    Information about the project transactions
+
+    """
+
     project_id: int
     org_id: int
-    transaction_counts: List[Tuple[str, int]]
+    transaction_counts: List[Tuple[str, float]]
+    total_num_transactions: Optional[float]
+    total_num_classes: Optional[int]
+
+
+class ProjectTransactionsTotals(TypedDict, total=True):
+    project_id: int
+    org_id: int
+    total_num_transactions: float
+    total_num_classes: int
+
+
+def same_project(left: Optional[ProjectIdentity], right: Optional[ProjectIdentity]) -> bool:
+    if left is None and right is None:
+        return True  # both None
```

Comment:
it's not clear why do you need True here, None != None, it's not the same project.

Reply:
also you have it here:
```
    if left is not None and right is not None and not same_project(left, right):

```

Reply:
@andriisoldatenko you are right, I could return False as well ( doesn't really matter) as long as one is None I don't want to continue with the comparison so I'll change the first two tests with:.
```
    if left is None or right is None:
        return False  

```

Reply:
The second test is needed:

```
  if left is not None and right is not None and not same_project(left, right):
```

 I only want to throw an error in `merge_transactions` if both are non None and different.
The function should work fine with one None and the other not None,